packages/engine/paima-sm/src/index.ts

```diff
@@ -224,20 +230,43 @@ const SM: GameStateMachineInitializer = {
 
         await processInternalEvents(latestChainData.internalEvents, dbTx);
 
+        // Used to disambiguate when two primitives have events in the same tx. This
+        // means the hash depends on the order we process the primitives (which is the
+        // order in the configuration).
+        // note that events triggered by the same tx may not be consecutive.
```

Comment:
I haven't really tested if it's the case since it's not that easy to setup, so I may be missing something, but as far as I understand from the current code we only group things by block at the funnel level. 

This means you can have 3 extensions that get triggered in the same block. And if two of them trigger in the same transaction, but the one that does not is in the middle of them (in the configuration order), the events here would also follow that order.

Reply:
I mostly added this comment because I initially had written the logic assuming that though, it's probably not really relevant.

Reply:
Ooohh, I get it now

I read this as "may not" → cannot possibly happen (aka "shall not")
but you meant "may not" → there is a chance it won't happen

Reply:
>I read this as "may not" → cannot possibly happen (aka "shall not")

I actually didn't even know that was actual english. I could rephrase to `don't need` I guess.

Reply:
I don't think it's likely for that to happen in practice to be honest, but it's a bit counter-intuitive.